### ISSUE TITLE
Prefer "Explicit resource management"

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#2649]
### Description

Not sure what kind of patterns we should catch. But I guess, if a variable is only used in `try` and called a method in `finally` can definitely replaced with `using`

### Examples

```js
// ❌
const foo = something();

try {
  // ...
} finally {
  foo.close()
}

// ✅
{
  using foo = something();
  // ...
}
```

```js
// ❌
let foo

try {
  // ...
} finally {
  foo?.destory()
}

// ✅
{
  using foo = something();
  // ...
}
```

```js
// ❌
const foo = something();
let bar;

try {
  // ...
} finally {
  foo.destory()
  bar.shutdown()
}

// ✅
{
  using foo = something();
  using bar = somethingElse();
  // ...
}
```

### Proposed rule name

`prefer-using` ? `prefer-dispose`?

### Additional Info

[Explicit resource management](https://tc39.es/proposal-explicit-resource-management/)

**Comment by fisker** [comment]
A class with some method like `close`/`destroy` maybe should be reported too.

**Comment by fisker** [comment]
Explicit Resource Management has (conditionally) advanced to Stage 4 https://x.com/rbuckton/status/1927744228576129247
